ktor-utils/jvm/src/io/ktor/util/EncodersJvm.kt

```diff
@@ -48,11 +48,32 @@ private fun CoroutineScope.inflate(
         val header = source.readPacket(GZIP_HEADER_SIZE)
         val magic = header.readShortLittleEndian()
         val format = header.readByte()
-        val padding = header.readBytes()
+        val flags = header.readByte().toInt()
+        // val time = header.readInt()
+        // val extraFlags = header.readByte()
+        // val osType = header.readByte()
+
+        /* flags bits:
+            bit 0   FTEXT
+            bit 1   FHCRC
+            bit 2   FEXTRA
+            bit 3   FNAME
+            bit 4   FCOMMENT
+            bit 5   reserved
+            bit 6   reserved
+            bit 7   reserved
+        */
+        if (flags and 0b0000_0100 != 0) { //FLG.FEXTRA
+            val extraLen = source.readShort().toInt()
+            val extra = source.readPacket(extraLen)
```

Comment:
I suspect this is here only to skip past the extra packet, a comment would help make this visible.

Reply:
I did some refactoring to make the code even more readable and added a comment as proposed.